bundles/org.openhab.binding.sonyprojector/src/main/java/org/openhab/binding/sonyprojector/internal/handler/SonyProjectorHandler.java

```diff
@@ -0,0 +1,692 @@
+import org.openhab.binding.sonyprojector.internal.communication.SonyProjectorStatusPower;
+import org.openhab.binding.sonyprojector.internal.communication.sdcp.SonyProjectorSdcpConnector;
+import org.openhab.binding.sonyprojector.internal.communication.sdcp.SonyProjectorSdcpSimuConnector;
+import org.openhab.binding.sonyprojector.internal.communication.serial.SonyProjectorSerialConnector;
+import org.openhab.binding.sonyprojector.internal.communication.serial.SonyProjectorSerialOverIpConnector;
+import org.openhab.binding.sonyprojector.internal.communication.serial.SonyProjectorSerialSimuConnector;
+import org.openhab.binding.sonyprojector.internal.configuration.SonyProjectorEthernetConfiguration;
+import org.openhab.binding.sonyprojector.internal.configuration.SonyProjectorSerialConfiguration;
+import org.openhab.binding.sonyprojector.internal.configuration.SonyProjectorSerialOverIpConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SonyProjectorHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Markus Wehrle - Initial contribution
+ * @author Laurent Garnier - Refactoring, poll thread for regular channels updates, new serial thing type, new channels
+ */
+@NonNullByDefault
+public class SonyProjectorHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SonyProjectorHandler.class);
+
+    private static final SonyProjectorModel DEFAULT_MODEL = SonyProjectorModel.VW520;
+    private static final long POLLING_INTERVAL = TimeUnit.SECONDS.toSeconds(15);
+
+    private @NonNullByDefault({}) ScheduledFuture<?> refreshJob;
+
+    private boolean identifyProjector;
+    private SonyProjectorModel projectorModel = DEFAULT_MODEL;
+    private SonyProjectorConnector connector = new SonyProjectorSdcpSimuConnector(DEFAULT_MODEL);
+
+    private SonyProjectorStateDescriptionOptionProvider stateDescriptionProvider;
+    private SerialPortManager serialPortManager;
+
+    private boolean simu;
+
+    private final Object commandLock = new Object();
+
+    public SonyProjectorHandler(Thing thing, SonyProjectorStateDescriptionOptionProvider stateDescriptionProvider,
+            SerialPortManager serialPortManager) {
+        super(thing);
+        this.stateDescriptionProvider = stateDescriptionProvider;
+        this.serialPortManager = serialPortManager;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String channel = channelUID.getId();
+        if (command instanceof RefreshType) {
+            logger.debug("REFRESH command from channel {} is ignored; will be handled by next projector state poll",
+                    channel);
+            return;
+        }
+        if (getThing().getStatus() != ThingStatus.ONLINE) {
+            logger.debug("Thing is not ONLINE; command {} from channel {} is ignored", command, channel);
```

Comment:
Is it ensured that we only have OFFLINE status if there is really no connection available/possible? Otherwise, we usually do not do such a logic, but try the communication nonetheless.

Reply:
Status is set to OFFLINE in two cases: configuration error detected during thing handler initialization and later when open() failed during the data poll (every 15 seconds).
My idea was, I think, why letting the user sending commands while we were not able to poll data just before.
But if this looks too much weird, I can suppress this test.

Reply:
Well, you might then be OFFLINE, because there was a network hiccup 10 seconds ago during the last poll.  But that might have been temporary and the next poll could be successful again - so dropping a command that you just received might be the wrong decision.

Reply:
Well, that is true. I will suppress the test.

Reply:
Done